### PR TITLE
Prepare for prototype removal

### PR DESCRIPTION
**Pull request by rsandell**
See https://www.jenkins.io/blog/2023/05/12/removing-prototype-from-jenkins/

### Testing done
Manual testing done:
 * Created a domain and added credentials to it
 * Created a new credential from the "Add" button on a Git SCM configuration page.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by rsandell** [comment]
This will merge conflict with #481

**Review by raul-arabaolaza — APPROVED** [review]
I have confirmed the issues by trying to add a new domain, there is a very clear error message on my js console about `$` that does not happen when you use a jenkins version that still contains prototype.

The concrete fixes look good to me

**Comment by ampuscas** [comment]
I have tested it: 
In a Jenkins which has `prototype` removed, there are errors: 
Screenshot

When I start Jenkins and take into account this PR, there are no errors, and I can create a new domain and add credentials, without any errors

**Comment by rsandell** [comment]
I've reverted the changes to `select.js` so now creating a credential from for example a Git SCM credentials input on a job configuration page works again.

**Review by ampuscas — APPROVED** [review]

**Review by raul-arabaolaza — APPROVED** [review]

**Review by yaroslavafenkin — APPROVED** [review]
Looks fine security wise

**Comment by Wadeck** [comment]
Potentially introduced [JENKINS-72364](https://issues.jenkins.io/browse/JENKINS-72364)

File: src/main/resources/lib/credentials/dialog.js [hunk 1]
```diff
@@ -3,5 +3,5 @@ Behaviour.specify("#credentials-add-submit", 'credentials-dialog-add', 0, functi
 });
 
 Behaviour.specify("#credentials-add-abort", 'credentials-dialog-abort', 0, function (e) {
-    e.onclick = (_) => window.credentials.dialog.hide();
+    e.onclick = (_) => window.credentials.dialog.style.display = "none";
```
Comment (basil) [inline src/main/resources/lib/credentials/dialog.js:6]: This change appears to be in error, as this was YUI code, not Prototype code.

**Pull request merged**